core/controllers/admin_test.py

```diff
@@ -3036,3 +3056,87 @@ def test_update_blog_post_with_correct_params(self) -> None:
                 'published_on': '05/09/2000'
             },
             csrf_token=csrf_token)
+
+
+class GenerateDummyBlogTest(test_utils.GenericTestBase):
+    """Tests Generate Dummy data"""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+
+    def test_cannot_generate_dummy_blog_in_prod_mode(self) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+
+        prod_mode_swap = self.swap(constants, 'DEV_MODE', False)
+        assert_raises_regexp_context_manager = self.assertRaisesRegex(
+            Exception, 'Cannot load blog in production.')
+
+        with assert_raises_regexp_context_manager, prod_mode_swap:
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_blog',
+                    'blog_title': 'Education',
+                }, csrf_token=csrf_token)
+
+        blog_count = (
+            blog_services.get_total_number_of_published_blog_post_summaries()
+        )
+        self.assertEqual(blog_count, 0)
+        self.logout()
+
+    def test_generate_dummy_blog(self) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+        self.post_json(
+            '/adminhandler', {
+                'action': 'generate_dummy_blog',
+                'blog_title': 'Education',
+            }, csrf_token=csrf_token)
+        blog_count = (
+            blog_services.get_total_number_of_published_blog_post_summaries()
+        )
+        self.assertEqual(blog_count, 1)
+
+        self.post_json(
+            '/adminhandler', {
+                'action': 'generate_dummy_blog',
+                'blog_title': 'Testing types of Text formatting',
+            }, csrf_token=csrf_token)
+        blog_count = (
+            blog_services.get_total_number_of_published_blog_post_summaries()
+        )
+        self.assertEqual(blog_count, 2)
+
+        self.post_json(
+            '/adminhandler', {
+                'action': 'generate_dummy_blog',
+                'blog_title': 'Leading The Arabic Translations Team',
+            }, csrf_token=csrf_token)
+        blog_count = (
+            blog_services.get_total_number_of_published_blog_post_summaries()
+        )
+        self.assertEqual(blog_count, 3)
+        self.logout()
+
+    def test_handler_raises_error_with_invalid_blog_title(self) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+        response = self.post_json(
+            '/adminhandler', {
+                'action': 'generate_dummy_blog',
+                'blog_title': 'blog title',
+            }, csrf_token=csrf_token, expected_status_int=400)
+
+        error_msg = (
+                        'The \"blog_title\" is not valid. It should be '
```

Comment:
This and the following should be indented by 4 from previous line (starting from `error_msg`), similar to self.post_json(... above